pkg/cmd/multiget.go

```diff
@@ -0,0 +1,467 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubectl-multi/pkg/cluster"
+	"kubectl-multi/pkg/util"
+)
+
+// ClusterInfo for multiget (may include extra fields for ITS)
+type MultiGetClusterInfo struct {
+	Name           string
+	KubeconfigPath string
+	Client         *kubernetes.Clientset
+	DynamicClient  dynamic.Interface
+	RestConfig     *rest.Config
+}
+
+func toClusterInfo(m MultiGetClusterInfo) cluster.ClusterInfo {
+	return cluster.ClusterInfo{
+		Name:          m.Name,
+		Context:       m.Name, // Use ITS name as context
+		Client:        m.Client,
+		DynamicClient: m.DynamicClient,
+		RestConfig:    m.RestConfig,
+	}
+}
+
+func newMultiGetCommand() *cobra.Command {
+	var outputFormat string
+	var selector string
+	var showLabels bool
+	var watch bool
+	var watchOnly bool
+
+	cmd := &cobra.Command{
+		Use:   "multiget [TYPE[.VERSION][.GROUP] [NAME | -l label] | TYPE[.VERSION][.GROUP]/NAME ...]",
+		Short: "Display resources across all ITS clusters discovered from Kubestellar core",
+		Long:  `Get resources from all ITS clusters by discovering them from the Kubestellar core cluster. No dependency on kflex or pre-imported kubeconfigs.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return fmt.Errorf("resource type must be specified")
```

Comment:
The "Use" message has square brackets around all the args, which means that it is valid to invoke this with no args.